pycalphad/tests/test_equilibrium.py

```diff
@@ -911,6 +911,49 @@ def test_MQMQA_ternary_equilibrium_ideal(load_database):
     assert np.allclose(eq.Y.values.squeeze()[0, :], [0.25000, 0.20000, 0.30000, 4.0000E-02, 0.12000, 9.0000E-02], atol=1e-5)  # Thermochimica result
 
 
+@select_database("MQMQA-tern-tests.dat")
+def test_MQMQA_equilibrium_binary_excess_same_chemical_groups(load_database):
+    """Binary excess terms with the same chemical group"""
+    dbf = load_database()
+    comps = ['CU', 'NI', 'VA']
+    eq = equilibrium(dbf, comps, ['XS_SAME_CG'], {v.P: 101325, v.T: 1000, v.N: 1, v.X('NI'): 0.5})
+    print('GM', eq.GM.values.squeeze())
+    print('Y', eq.Y.values.squeeze())
+    print('Phase', eq.Phase.values.squeeze())
+    assert np.isclose(eq.GM.values.squeeze(), -9.29581E+03)  # Thermochimica result
+    assert np.all(eq.Phase.squeeze() == ['XS_SAME_CG', '', ''])
+    assert np.allclose(eq.Y.values.squeeze()[0, :], [0.12442, 0.50078, 0.37481], atol=1e-5)  # Thermochimica result
+
+
+@pytest.mark.solver
+@pytest.mark.xfail
+@select_database("Be-F-Li.dat")
+def test_MQMQA_species_of_different_moles(load_database):
+    """Ternary ideal"""
+    dbf = load_database()
+    comps = ['F', 'LI', 'BE', 'VA']
+    eq = equilibrium(dbf, comps, ['MSFL'], {v.N: 1, v.P: 101325, v.T: 1200, v.X('LI'):0.2, v.X('BE'):0.2}, calc_opts={'pdens': 50000})
+    print('GM', eq.GM.values.squeeze())
+    print('Y', eq.Y.values.squeeze())
+    print('Phase', eq.Phase.values.squeeze())
+    assert np.isclose(eq.GM.values.squeeze(), -3.75743E+05)  # Thermochimica result
+    assert np.all(eq.Phase.squeeze() == ['MSFL', '', '', ''])
+    assert np.allclose(eq.Y.values.squeeze()[0, :],[5.4835E-05, 6.8482E-03, 2.6583E-02, 0.21381, 0.65434, 9.8365E-02], atol=1e-5)  # Thermochimica result
```

Comment:
This one is interesting. The test does indeed fail (it's marked xfail), but only the site fractions don't match. We get the right energy and phase. Could you double check the Thermochimica values?

Reply:
I also removed the pdens override and get virtually the same answer within PyCalphad down to <5 µJ.